idseq_pipeline/commands/non_host_alignment_functions.py

```diff
@@ -124,17 +124,15 @@ def result_dir(basename):
 
 # convenience functions
 def count_lines(input_file):
-    return int(
-        execute_command_with_output(
-            "wc -l %s" % input_file).strip().split()[0])
+    cmd = "wc -l %s" % input_file
+    return int(execute_command_with_output(cmd).strip().split()[0])
 
 
 def count_lines_in_paired_files(input_files):
     distinct_counts = list(set(map(count_lines, input_files)))
-    assert len(
-        distinct_counts
-    ) == 1, "Mismatched line counts in supposedly paired files: {}".format(
+    msg = "Mismatched line counts in supposedly paired files: {}".format(
         input_files)
+    assert len(distinct_counts) == 1, msg
```

Comment:
There is no need to compute the message if the assert condition passes, which is the common case.   In this refactor, however, the message is always computed.   I am curious if python includes the optimization I am assuming exists.